### PR TITLE
Ap 3399 call CFE for some bank statement journey cases

### PR DESCRIPTION
**Pull request by willc-work**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-3399)

When an employed applicant with a HMRC response uses bank statement upload journey their employment data is assessed by CFE

Add a check for state of application and user privileges
add a call to CFE when necessary

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Review by RoseSAK — APPROVED** [review]
Looks good 💯

**Review by jsugarman — APPROVED** [review]
LGTM 👍 

be good to get the means report chages in and put this on top

File: app/views/providers/means_reports/_without_cfe_result_details.erb [hunk 1]
```diff
@@ -10,7 +18,9 @@
   </section>
 
   <section class="print-no-break">
-    <%= render('income_cash_payments', read_only: true) %>
+    <% if @legal_aid_application.legal_aid_application_transaction_types.any? %>
```
Comment (jsugarman) [inline app/views/providers/means_reports/_without_cfe_result_details.erb:21]: i would query whether we want to display "None" here if such instances

File: app/views/providers/means_reports/_without_cfe_result_details.erb [hunk 2]
```diff
@@ -22,7 +32,9 @@
   </section>
 
   <section class="print-no-break">
-    <%= render('outgoings_cash_payments', read_only: true) %>
+    <% if @legal_aid_application.legal_aid_application_transaction_types.any? %>
```
Comment (jsugarman) [inline app/views/providers/means_reports/_without_cfe_result_details.erb:35]: i would query whether we want to display "None" here if such instances

**Pull request merged**